credentials/src/main/kotlin/web5/sdk/credentials/StatusListCredential.kt

```diff
@@ -0,0 +1,317 @@
+
+    val statusListCredStatusPurpose: String? =
+      statusListCredential.vcDataModel.credentialSubject.jsonObject[STATUS_PURPOSE] as? String?
+
+    if (statusListEntryValue.statusPurpose == null) {
+      throw IllegalArgumentException("Status purpose in the credential to validate is null")
+    }
+
+    if (statusListCredStatusPurpose == null) {
+      throw IllegalArgumentException("Status purpose in the status list credential is null")
+    }
+
+    if (statusListEntryValue.statusPurpose != statusListCredStatusPurpose) {
+      throw IllegalArgumentException("Status purposes do not match between the credentials")
+    }
+
+    val compressedBitstring: String? =
+      statusListCredential.vcDataModel.credentialSubject.jsonObject[ENCODED_LIST] as? String?
+
+    if (compressedBitstring == null || compressedBitstring.isEmpty()) {
+      throw IllegalArgumentException("Compressed bitstring is null or empty")
+    }
+
+    val credentialIndex = statusListEntryValue.statusListIndex
+    val expandedValues: List<String> = bitstringExpansion(compressedBitstring)
+
+    return expandedValues.any { it == credentialIndex }
+  }
+
+
+  /**
+   * Validates if a given credential is part of the status list.
+   *
+   * @param credentialToValidate The [VerifiableCredential] to be validated against the status list.
+   * @param httpClient An optional [HttpClient] for fetching the status list credential. If not provided, a default HTTP client will be used.
+   * @return A [Boolean] indicating whether the `credentialToValidate` is part of the status list.
+   *
+   * This function fetches the status list credential from a URL present in the `credentialToValidate`.
+   * It supports using either a user-provided `httpClient` or a default client when no client is passed in.
+   * The function then checks if the given `credentialToValidate`'s status list index is present in the expanded status list derived from the fetched status list credential.
+   *
+   * Example:
+   * ```
+   * val isRevoked = validateCredentialInStatusList(credentialToCheck)
+   * ```
+   */
+  public suspend fun validateCredentialInStatusList(
+    credentialToValidate: VerifiableCredential,
+    httpClient: HttpClient? = null // default HTTP client but can be overridden
+  ): Boolean {
+
+    var isDefaultClient = false
+    val clientToUse = httpClient ?: defaultHttpClient().also { isDefaultClient = true }
+
+    try {
+      val statusListEntryValue: StatusList2021Entry =
+        StatusList2021Entry.fromJsonObject(credentialToValidate.vcDataModel.credentialStatus.jsonObject)
+      val statusListCredential =
+        clientToUse.fetchStatusListCredential(statusListEntryValue.statusListCredential.toString())
+
+      return validateCredentialInStatusList(credentialToValidate, statusListCredential)
+    } finally {
+      if (isDefaultClient) {
+        clientToUse.close()
+      }
+    }
+  }
+
+  private fun defaultHttpClient(): HttpClient {
+    return HttpClient(CIO)
```

Comment:
I think you'd probably want to set content negotiation with jackson, similar to https://github.com/TBD54566975/web5-kt/blob/6bc48b81e15a9e7be6316eda0bccf518c14e335b/dids/src/main/kotlin/web5/sdk/dids/DidIonManager.kt#L112